cmd/xl-storage-format-v2.go

```diff
@@ -1810,6 +1812,11 @@ func mergeXLV2Versions(quorum int, strict bool, versions ...[]xlMetaV2ShallowVer
 			}
 		}
 
+		// Calculate latest 'n' non-free versions.
+		if !latest.header.FreeVersion() {
+			nVersions++
+		}
```

Comment:
Only do this check in both `merged = append(merged, latest)` where we have quorum.